### PR TITLE
Port `test_pulsepacket_generator` from SLI-2-Py

### PR DESCRIPTION
**Pull request by nicolossus**
The PR also improves the formatting of the documentation for `stimulation_device` and `pulsepacket_generator`.

### CODE REVIEW

**Comment by nicolossus** [comment]
@jougs @jessica-mitchell The SLI-2-Py port is quite one-to-one and straightforward. However, in porting the test I noticed some places in the documentation that could be improved by some small changes to the formatting . So far I have only made similar changes to the documentation on files I have touched, see https://github.com/nest/nest-simulator/pull/2813. I will make the corresponding changes elsewhere once we agree on the style.

File: nestkernel/stimulation_device.h [hunk 1]
```diff
@@ -48,14 +48,15 @@ global offset that shifts the stimulation period. All three values are
...
 
 - For current-emitting devices, the current is activated and
   deactivated such that the current first affects the target dynamics
-  during the update step (start, start+h], i.e., an effect can be
-  recorded at the earliest at time start+h. The last interval during
-  which the current affects the target's dynamics is (stop-h, stop].
+  during the update step :math:`(\mathrm{start}, \mathrm{start}+h]`,
+  that is, an effect can be recorded at the earliest at time
+  :math:`\mathrm{start}+h`. The last interval during which the current
+  affects the target's dynamics is :math:`(\textrm{stop}-h, \texrm{stop}]`.
```
Comment (jessica-mitchell) [inline nestkernel/stimulation_device.h:59]:
```suggestion
  affects the target's dynamics is :math:`(\textrm{stop}-h, \textrm{stop}]`.
```

**Review by jessica-mitchell — CHANGES_REQUESTED** [review]
Just a minor change, 
and more to @jougs the stimulation_device userdocs in nestkernel has a NOINDEX label, and I don't think it's in Read the docs. I think this should be included somewhere, oder?

**Review by jessica-mitchell — APPROVED** [review]

File: testsuite/pytests/sli2py_stimulating/test_pulsepacket_generator.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    params = {"pulse_times": [], "activity": 0, "sdev": 0.0}
+    ppg = nest.Create("pulsepacket_generator")
+    ppg.set(params)
+    nest.Simulate(1.0)
+
+
+def test_statistical_correctness():
+    """
+    Test the statistical correctness of `pulsepacket_generator`.
+
```
Comment (heplesser) [inline testsuite/pytests/sli2py_stimulating/test_pulsepacket_generator.py:84]:
```suggestion
def test_number_of_spikes():
    """
    Test that `pulsepacket_generator` emits expected number of spikes.

```

File: testsuite/pytests/sli2py_stimulating/test_pulsepacket_generator.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Test parameter setting and statistical correctness of `pulsepacket_generator`.
```
Comment (heplesser) [inline testsuite/pytests/sli2py_stimulating/test_pulsepacket_generator.py:23]:
```suggestion
Test parameter setting and correct number of spikes emitted by `pulsepacket_generator`.
```

**Review by heplesser — APPROVED** [review]

**Pull request merged**